x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -99,7 +99,20 @@ field
     ;
 
 fromCommand
-    : FROM fromIdentifier (COMMA fromIdentifier)* metadata?
+    : FROM fromIdentifier (COMMA fromIdentifier)* fromOptions? metadata?
+    ;
+
+fromIdentifier
+    : FROM_UNQUOTED_IDENTIFIER
+    | QUOTED_IDENTIFIER
+    ;
+
+fromOptions
+    : OPTIONS configOption (COMMA configOption)*
+    ;
+
+configOption
+    : string ASSIGN string
```

Comment:
Indeed, ideally one would simply write smth like `OPTIONS allow_no_indices=true`. 
But I think the rational was that (1) it might require explaining when to use value quotes and when not (not that difficult, but would come with some load) and (2) that we can relax the quoting later, if needed.